DeepStorage/Deep_Storage.cs

```diff
@@ -343,6 +343,30 @@ public class CompDeepStorage : ThingComp {
 
 
         }
+
+        #region IHoldMultipleThings //PickUpAndHaul Compatibility
+        public bool CapacityAt(ThingDef def, IntVec3 storeCell, Map map, out int capacity)
+        {
+            ISlotGroupParent slotGroup = map.haulDestinationManager.SlotGroupParentAt(storeCell);
+            int maxStacks = (slotGroup as ThingWithComps)?.GetComp<CompDeepStorage>()?.maxNumberStacks ?? 0;
+
+            List<Thing> stuffHere = map.thingGrid.ThingsListAt(storeCell);
+
+            int max = maxStacks * def.stackLimit;
+
+            capacity = Math.Min(max, stuffHere.Where(x => x.def.EverStorable(false) && x.def == def)
+                                              .Select(x => x.def.stackLimit - x.stackCount)
+                                              .Sum());
+            return capacity > 0;
+        }
+
+        public bool StackableAt(ThingDef def, IntVec3 storeCell, Map map)
+            => (map.haulDestinationManager.SlotGroupParentAt(storeCell) as ThingWithComps)?.GetComp<CompDeepStorage>()?.maxNumberStacks > 0
+            && map.thingGrid.ThingsListAt(storeCell).Where(x => x.def.EverStorable(false) && x.def == def)
+                  .Select(x => x.def.stackLimit - x.stackCount).Any();
```

Comment:
This select statement turns all matching Things into an integer of how far they are from full?  And then returns true if there are any at all, no matter what the Select() did to them?